Opponent Play

rodney528 opened the pull request:
I noticed opponent play was commented option in ClientPrefs so I'm doing this now ig.

rodney528 committed:
Added isStagePixel and curStage for if statements

rodney528 committed:
Update FunkinLua.hx

rodney528 committed:
Update FunkinLua.hx

rodney528 committed:
Added some stuff to FunkinLua.hx

rodney528 committed:
Opps

I made it the health bar not the health amount.

rodney528 committed:
Added chartingMode

Name says it all but ex:
if chartingMode then
    -- Any thing here will only appear in charting mode!
end

rodney528 committed:
God dang it

getHealth already existed

rodney528 committed:
Update... yeah.

I think I needed to add instance? I really need to figure out how to compile to test this shit.

rodney528 committed:
Equal amount of binds

keyJustPressed, keyPressed and keyReleased now all have the ability to grab the same buttons!

rodney528 committed:
Merge branch 'main' of https://github.com/ImaginationSuperHero528/FNF-PsychEngine

rodney528 committed:
Added setOnLuas
Like the title says I added setOnLuas... cause why not?

rodney528 committed:
Forget the ; opps

rodney528 committed:
Small Change

Should probably be set as `Dynamic` and not `String`.

rodney528 committed:
Easy use of getProperty

Only trying to do ones that need getPropertyFromClass (I think that's one I wanna refer to?)

rodney528 committed:
I'm basically removing anything using getProperty

rodney528 committed:
If "dad.curCharacter" is same as "gf.curCharacter"

Basically ya know how in the tutorial gf is technically the dad? Well this makes it (I hope) so it can work with anyone other then just gf.

rodney528 committed:
playbackRate to duration stuff in FunkinLua

rodney528 committed:
did this too ig

rodney528 committed:
Did more of PlayState

rodney528 committed:
Made it if equals now

dad.curCharacter.startsWith(gf.curCharacter) has been replaced with dad.curCharacter == gf.curCharacter

rodney528 committed:
Still working on :ShadowTroll:

rodney528 committed:
:P

rodney528 committed:
For duration it should be / woops!

rodney528 committed:
Screw it, fuck if easy use

rodney528 committed:
Merge branch 'main' into playbackRate-to-duration-stuff-in-FunkinLua

rodney528 committed:
progress ig

rodney528 committed:
Changed up

rodney528 committed:
Uh... fixed?

rodney528 committed:
Revert "Uh... fixed?"

This reverts commit f6973c0b984068100144503994bd0a99044585d1.

rodney528 committed:
Revert "Changed up"

This reverts commit 8811146d59a06555bc5e36c6e00dc1f8a387fc05.

rodney528 committed:
Hopefully it's fixed now

rodney528 committed:
it's been a bit

thx kingyomoma

Co-Authored-By: kingyomoma <96543290+kingyomoma@users.noreply.github.com>

rodney528 committed:
:whyyyyy:

rodney528 committed:
Merge branch 'setOnLuas'

rodney528 committed:
Merge branch 'loadSong-FIX'

rodney528 committed:
Merge branch 'playbackRate-to-duration-stuff-in-FunkinLua'

rodney528 committed:
Merge branch 'if-dad-is-same-as-gf'

rodney528 committed:
Merge branch 'Lua-Adds'

rodney528 committed:
Revert "Merge branch 'Lua-Adds'"

This reverts commit b7f13616fe8fd5df604094eaeb9b62d31549d322, reversing
changes made to b4e8964711874b4f0ccc5eb9e1a9f37ad60762d3.

rodney528 committed:
Revert "Merge branch 'if-dad-is-same-as-gf'"

This reverts commit b4e8964711874b4f0ccc5eb9e1a9f37ad60762d3, reversing
changes made to 46c5d94eb1274b45cf9e3daa883a388bf8a8af74.

rodney528 committed:
Revert "Merge branch 'playbackRate-to-duration-stuff-in-FunkinLua'"

This reverts commit 46c5d94eb1274b45cf9e3daa883a388bf8a8af74, reversing
changes made to 7606d1597f246df37a613075a7c6d538916abf0a.

rodney528 committed:
Revert "Merge branch 'loadSong-FIX'"

This reverts commit 7606d1597f246df37a613075a7c6d538916abf0a, reversing
changes made to 2ad13aaa39ad9f4046263265097a1b4ec6f61dc5.

rodney528 committed:
Revert "Merge branch 'setOnLuas'"

This reverts commit 2ad13aaa39ad9f4046263265097a1b4ec6f61dc5, reversing
changes made to bdb6910951cc1a28c8c6a8898091789287f70004.

rodney528 committed:
Opponent Play

I noticed opponent play was a uncommented option in ClientPrefs so I'm doing this now ig.

rodney528 committed:
Merge branch 'main' into opponent-play

antoSKRLOL commented:
finally someone fixed it (I hope so)

UmbratheUmbreon commented on source/PlayState.hx:
```diff
@@ -1313,7 +1315,9 @@ class PlayState extends MusicBeatState
 		// Updating Discord Rich Presence.
-		DiscordClient.changePresence(detailsText, SONG.song + " (" + storyDifficultyText + ")", iconP2.getCharacter());
+		var setIcon:HealthIcon = iconP2;
+		if (opponentPlay) {setIcon = iconP1;} else {setIcon = iconP2;}
```
You could just do 
`var setIcon:HealthIcon = (opponentPlay ? iconP1 : iconP2);`

ImaginationSuperHero528 replied:
Oh... I didn't know that so thx!

DillyzThe1 commented:
hey i've actually done this in both kade and psych engine before (both times were a result of leftover code i had from making online multiplayer for my friends) so if you have any problems getting anything to work let me know

ImaginationSuperHero528 commented:
> hey i've actually done this in both kade and psych engine before (both times were a result of leftover code i had from making online multiplayer for my friends) so if you have any problems getting anything to work let me know

Alr thx

rodney528 committed:
Optimized... wait I forgot gcs substate shit

Thx UmbratheUmbreon for the tip!

Co-Authored-By: AT <101066547+UmbratheUmbreon@users.noreply.github.com>

rodney528 committed:
:P

rodney528 committed:
Last update for today.

rodney528 committed:
Update PlayState.hx

rodney528 committed:
Added Event

rodney528 committed:
probably should do this

rodney528 committed:
Merge remote-tracking branch 'upstream/main'

rodney528 committed:
Merge branch 'main' into opponent-play

rodney528 committed:
Revert "Merge branch 'main' into opponent-play"

This reverts commit 563ff739b68394e4748cfa1d370737997fbc8750, reversing
changes made to 8def79a990d7f402b343060ce09969c9ae624c5b.

RodneyAnImaginativePerson commented:
> finally someone fixed it (I hope so)

Wdym fixed?

rodney528 committed:
woops

rodney528 committed:
Should run nicely... I hope

rodney528 committed:
Update PlayState.hx

rodney528 committed:
ugh

rodney528 committed:
Update PlayState.hx

rodney528 committed:
I forgot about some if checks lol

rodney528 committed:
update today

I broke player sustains when oppo play off... hehe

rodney528 committed:
update PlayState.hx

rodney528 committed:
update PlayState.hx

rodney528 committed:
update PlayState.hx

rodney528 committed:
update PlayState.hx

rodney528 committed:
update PlayState.hx

rodney528 committed:
update PlayState.hx

rodney528 committed:
update PlayState.hx

rodney528 committed:
update PlayState.hx

rodney528 committed:
update PlayState.hx

rodney528 committed:
update PlayState.hx

rodney528 committed:
update PlayState.hx

rodney528 committed:
update PlayState.hx

rodney528 committed:
update PlayState.hx

rodney528 committed:
update PlayState.hx

rodney528 committed:
update PlayState.hx

rodney528 committed:
update PlayState.hx

rodney528 committed:
update PlayState.hx

rodney528 committed:
update PlayState.hx

rodney528 committed:
update PlayState.hx

rodney528 committed:
update PlayState.hx

rodney528 committed:
update PlayState.hx

rodney528 committed:
update PlayState.hx

rodney528 committed:
update PlayState.hx

rodney528 committed:
update PlayState.hx

rodney528 committed:
update PlayState.hx

rodney528 committed:
update PlayState.hx

rodney528 committed:
update PlayState.hx

rodney528 committed:
update PlayState.hx

rodney528 committed:
update PlayState.hx

rodney528 committed:
update PlayState.hx

rodney528 committed:
update PlayState.hx

rodney528 committed:
update PlayState.hx

rodney528 committed:
update PlayState.hx

rodney528 committed:
update PlayState.hx

rodney528 committed:
update PlayState.hx

rodney528 committed:
update PlayState.hx

rodney528 committed:
update PlayState.hx

rodney528 committed:
update PlayState.hx

rodney528 committed:
update PlayState.hx

rodney528 committed:
update PlayState.hx

rodney528 committed:
update PlayState.hx

rodney528 committed:
update PlayState.hx

rodney528 committed:
update PlayState.hx

rodney528 committed:
update PlayState.hx

rodney528 committed:
update PlayState.hx

rodney528 committed:
update PlayState.hx

rodney528 committed:
update PlayState.hx

rodney528 committed:
update PlayState.hx

rodney528 committed:
update PlayState.hx

rodney528 committed:
update PlayState.hx

rodney528 committed:
update PlayState.hx

rodney528 committed:
update PlayState.hx

rodney528 committed:
update PlayState.hx

rodney528 committed:
update PlayState.hx

rodney528 committed:
update PlayState.hx

rodney528 committed:
update PlayState.hx

rodney528 committed:
update PlayState.hx

rodney528 committed:
update PlayState.hx

rodney528 committed:
update PlayState.hx

rodney528 committed:
update PlayState.hx

rodney528 committed:
update PlayState.hx

RodneyAnImaginativePerson commented:
JESUS

rodney528 committed:
update PlayState.hx

rodney528 committed:
update PlayState.hx

rodney528 committed:
update PlayState.hx

RodneyAnImaginativePerson commented:
I didn't think everytime I saved pushed my damn work!

RodneyAnImaginativePerson commented:
For y'all wondering I was coding on my phone lol

rodney528 committed:
doing shit

rodney528 committed:
I GOT IT WORKING KINDA NOW

rodney528 committed:
Why was noteData being rounded when it wasn't needed???

rodney528 committed:
Getting Better!

Only issues now are not being able to hit opponent sustains and bf not hitting notes when playing as the opponent.

rodney528 committed:
last update for today

rodney528 committed:
update PlayState.hx

rodney528 committed:
update PlayState.hx

rodney528 committed:
update PlayState.hx

rodney528 committed:
update PlayState.hx

rodney528 committed:
update PlayState.hx

rodney528 committed:
update PlayState.hx

rodney528 committed:
update PlayState.hx

rodney528 committed:
Update PlayState.hx

rodney528 committed:
Merge remote-tracking branch 'upstream/main'

rodney528 committed:
Merge remote-tracking branch 'upstream/main' into opponent-play

rodney528 committed:
Merge branch 'main' into opponent-play

rodney528 committed:
Basically done

Just gotta fix the arrow alpha when opponent play on start.

RodneyAnImaginativePerson commented:
[VIDEO](https://www.youtube.com/watch?v=lG9UakC4c1I) :D

rodney528 committed:
Some fixes/adjustments

didn't fix alpha problems yet

rodney528 committed:
hmmmm

rodney528 committed:
IS... DONE, VIDEO SOON BABY!!!

rodney528 committed:
ok we good now

rodney528 committed:
uhhhhhhh

rodney528 committed:
FORGOT ABOUT GIRLFRIEND

rodney528 committed:
Slowly removing unnecessary changes

rodney528 committed:
ugh

rodney528 committed:
omg

rodney528 committed:
oh boy

rodney528 committed:
done with that now

rodney528 committed:
Last push for today... probably

rodney528 committed:
Merge remote-tracking branch 'upstream/main'

RodneyAnImaginativePerson commented:
Might move to the experimental branch!

rodney528 committed:
HOW

rodney528 committed:
Merge branch 'main' into opponent-play

charlesisfeline reviewed (changes requested):
you forgot something

```source/PlayState.hx:4757: characters 2-8 : Expected }```

charlesisfeline reviewed (changes requested):
you forgot something

```source/PlayState.hx:4757: characters 2-8 : Expected }```

the compiler is showing a compiler error from a completely unrelated line

rodney528 committed:
Create lua.booster.lint.json

rodney528 committed:
:sadness:

RodneyAnImaginativePerson commented:
> you forgot something
> 
> `source/PlayState.hx:4757: characters 2-8 : Expected }`
> 
> the compiler is showing a compiler error from a completely unrelated line

idk whats wrong give me 20 years

doom699 commented:
time to play opposition and break the fingers

RodneyAnImaginativePerson commented:
HEY I NEED TO PORT TO 0.7 STILL >:(

RodneyAnImaginativePerson commented:
> > HEY I NEED TO PORT TO 0.7 STILL >:(
> 
> he closed 99% of pre-0.7 PRs

Ah
